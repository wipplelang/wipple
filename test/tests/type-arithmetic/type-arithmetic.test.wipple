-- [should error]

Z : type
S : N => type

Addition : A B Sum => trait ()
A => instance (Addition Z A A) : ()
A B Sum where (Addition A B Sum) => instance (Addition (S A) B (S Sum)) : ()

result :: A where (Addition (S (S (S Z))) (S (S Z)) A) => A
result : ...

_ : _ -> result
