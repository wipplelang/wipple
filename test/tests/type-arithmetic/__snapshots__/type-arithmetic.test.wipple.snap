// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`type-arithmetic.test.wipple 1`] = `
[
  "/Users/wilson/Desktop/wipple-wasm/test/tests/type-arithmetic/type-arithmetic.test.wipple:12:4: error: this code produces a \`_ -> (S (S (S (S (S Z)))))\`, but the \`_\`s are unknown",
]
`;
